### PR TITLE
Implement nav with transitions

### LINKED ISSUE #35: Dashboard Nav should be delightfully animated and responsive
#### Perceived Problem

- Dashboard navigation looks great on wider screens but currently lacks responsiveness for mobile screens or even smaller laptops
- Users may want to have a totally minimalistic writing experience without being distracted by the wider nav
- We have some parts of the dashboard, such as `settings`, that actually require a narrower nav bar (see design B)

#### Ideas / Proposed Solution(s)

- Create a responsive system that allows users to manually collapse & expand the nav
- It should expand and collapse automatically based on screen width, while still being able to be toggled by the user
- Collapsing and expanding the nav should be delightfully and simply animated 👌🏼😌
- This isn't currently present in the designs, but it might be nice to have a nice double-chevron in our `brandBlue` color that allows them to manually toggle it open and closed. Just as a thought. Notion has a nice implementation of this! (see design C)

#### Designs

**A** - Here's how the nav looks at full width
image

**B** - Here's how it looks collapsed
image

**C** Here's how Notion has that nice little double chevron toggle:
image
image

### PR DESCRIPTION
**Pull request by nlitwin**
## Description

Resolves #35. This PR finishes styling the nav and adds transitions. See sub tasks for details. Follow up task for collapsing/expanding the nav manually in issue #57.

## Sub-Tasks

- [x] I have added this PR to the Journaly Kanban project ✅
- [x] Display background overlay on mobile only
- [x] Animate open and closed on mobile
- [x] Expose header hamburger menu icon above mobile nav for toggling
- [x] Transition between mobile and tablet smoothly
- [x] Show skinny nav for settings page, even on desktop
- [x] Transition dashboard content in sync with nav
- [x] Prevent the nav from transitioning on page load
- [x] Extract link text to translated strings

## Screenshots

fully_animated_nav

### CODE REVIEW

**Comment by robin-macpherson** [comment]
Awesome work man!!!! After pulling down the code and taking it for a spin it looks amazing, super super nice!! The one thing missing for now that might be worth adding before we merge this is the active dashboard link state. What do you think, would that be pretty easy to add? 

Now I'll move onto looking at the code!

File: frontend/components/Dashboard/Header/Header.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  onMenuClick: () => void
+}
+
+const Header: React.FC<Props> = ({ onMenuClick }) => {
+  return (
+    <div className="header">
+      <div className="hamburger-icon" onClick={onMenuClick}>
+        <div />
+        <div />
+        <div />
```
Comment (robin-macpherson) [inline frontend/components/Dashboard/Header/Header.tsx:15]: haha these divs look funny 😆

**Comment by robin-macpherson** [comment]
image

**Comment by robin-macpherson** [comment]
https://media.giphy.com/media/cFkiFMDg3iFoI/giphy.gif

File: frontend/components/Dashboard/Header/Header.tsx (hunk 1 above)
Comment (nlitwin) [reply frontend/components/Dashboard/Header/Header.tsx:15]: hahaha, prettier won't let me put them all in a line 😄 but i like that the DOM nodes actually visually look like a hamburger menu 😆

**Comment by nlitwin** [comment]
@robin-macpherson bahaha that gif! sounds good, I'll update the font 👍

**Pull request merged**